.github/workflows/static_analysis.yaml

```diff
@@ -39,33 +35,50 @@ jobs:
         #  --max-configs=6  :   Using less configuration permutations (default is 12) to reduce run time. Detects less errors. Removed.
         #  -Dxxx            :   preprocessor configuration to use. Relevant flags taken from build on Ubuntu.
         run: > 
-          cppcheck ./src ./include ./common ./tools ./examples ./third-party/realdds ./third-party/utilities
+          cppcheck ./src ./include ./common ./tools ./examples ./third-party/realdds ./third-party/rsutils
           --quiet --std=c++11 --xml -j4 --enable=warning
-          -I./src -I./include -I./third-party/utilities/include -I./common
+          -I./src -I./include -I./third-party/rsutils/include -I./common
           -i./src/mf -i./src/uvc -i./src/win -i./src/winusb --suppress=*:third-party/json.hpp
           -DBUILD_WITH_DDS -DHWM_OVER_XU -DRS2_USE_V4L2_BACKEND -DUNICODE -DUSING_UDEV -DCHECK_FOR_UPDATES -D__linux__
           &> cppcheck_run.log
 
-      - name: Upload log
+      - name: Diff
+        id: diff-step
+        continue-on-error: true 
+        shell: bash
+        run: |
+          python3 .github/workflows/cppcheck-parse.py --severity E cppcheck_run.log | sort --key 3 > cppcheck_run.parsed.log
+          python3 .github/workflows/cppcheck-parse.py --severity E .github/workflows/cppcheck_run.log | sort --key 3 > cppcheck_run.parsed.golden
+          diff cppcheck_run.parsed.golden cppcheck_run.parsed.log
+
+      - name: Upload logs
         uses: actions/upload-artifact@v3
         with: 
           name: cppcheck_log
-          path: cppcheck_run.log
-
+          path: |
+            cppcheck_run.log
+            cppcheck_run.parsed.log
+  
       - name: Provide correct exit status
         shell: bash
         run: |
-          ERROR_COUNT=$(grep cppcheck_run.log -e "severity=\"error\"" -c);
-          if [ $ERROR_COUNT -eq ${{env.CPPCHECK_EXPECTED_ERROR_COUNT}} ];
+          ERROR_COUNT=$(grep cppcheck_run.log -e "severity=\"error\"" -c)
+          EXPECTED_ERROR_COUNT=$(grep .github/workflows/cppcheck_run.log -e "severity=\"error\"" -c)
```

Comment:
Shouldn't expected be parsed from golden?

Reply:
Ahhh... I see that the file you saved is not named golden. Change it's name to something with "gold", or "expected" in it

Reply:
It's the name of the file that's the same as the artifact.
I needed the "golden" to distinguish the two generated "parsed" files, that's all.

Reply:
That way, you can simply copy the artifact on top of the original and not have to change any filenames etc.